File: app/src/main/java/com/narcyber/mvpbasics/presenter/AllUsersFragmentPresenter.java
package com.narcyber.mvpbasics.presenter;

import com.narcyber.mvpbasics.model.User;

import java.util.List;

public class AllUsersFragmentPresenter extends ParentPresenter {

    private final AllUserView view;

    public AllUsersFragmentPresenter(AllUserView view) {
        this.view = view;
    }

    public void getAllUsers() {
        getUserRepository().requestForAllUsers();
    }

    @Override
    public void respondForAllUsers(List<User> users) {
        view.allUsers(users);
    }

    public interface AllUserView {
        void allUsers(List<User> userList);
    }

}

File: app/src/main/java/com/narcyber/mvpbasics/presenter/AllUsersFragmentPresenter.kt
package com.narcyber.mvpbasics.presenter

import com.narcyber.mvpbasics.model.User

class AllUsersFragmentPresenter(val view: AllUserView) : ParentPresenter() {

    fun getAllUsers() {
        userRepository.requestForAllUsers()
    }

    override fun respondForAllUsers(users: List<User?>?) {
        view.allUsers(users)
    }

    interface AllUserView {
        fun allUsers(userList: List<User?>?)
    }
}